tests/unit/tasks_service.test.ts

```diff
@@ -2,18 +2,18 @@ import * as activityMock from "../mock/rest/activity";
 import { Task, TaskQueue, TaskService, Worker } from "../../src/task";
 import { agreementPoolServiceMock, paymentServiceMock, networkServiceMock, LoggerMock, YagnaMock } from "../mock";
 import { Result } from "../../src";
-let queue;
+let queue: TaskQueue<Task>;
 const logger = new LoggerMock();
 
 describe("Task Service", () => {
   beforeEach(() => {
     logger.clear();
     activityMock.clear();
-    queue = new TaskQueue<Task<any, any>>();
+    queue = new TaskQueue<Task>();
   });
   it("should process new task in queue", async () => {
-    const worker: Worker<null, Result> = async (ctx) => ctx.run("some_shell_command");
-    const task = new Task<null, Result>("1", worker);
+    const worker: Worker<Result> = async (ctx) => ctx.run("some_shell_command");
+    const task = new Task<Result>("1", worker);
```

Comment:
Suggestion: Write code to promote type inference. The equivalent would be:

```ts
const worker = async (ctx: WorkContext) => ctx.run("some_shell_command");
const task = new Task("1", worker);
```

This way We don't have to specify `Worker<Result>` | `Task<Result>` and the type inference does the job.